changes/tasks.py

```diff
@@ -545,14 +545,15 @@ def switch_channel(self, change):
             elif change.data["channel"] == "sms" and "whatsapp" in short_name:
                 # Change any WhatsApp subscriptions to SMS
                 sbm_client.update_subscription(sub["id"], {"active": False})
-                messageset = messagesets_rev[re.sub("^whatsapp_", "", short_name)]
-                SubscriptionRequest.objects.create(
-                    identity=sub["identity"],
-                    messageset=messageset,
-                    next_sequence_number=sub["next_sequence_number"],
-                    lang=sub["lang"],
-                    schedule=sub["schedule"],
-                )
+                if "whatsapp_service_info" not in short_name:
+                    messageset = messagesets_rev[re.sub("^whatsapp_", "", short_name)]
+                    SubscriptionRequest.objects.create(
+                        identity=sub["identity"],
+                        messageset=messageset,
+                        next_sequence_number=sub["next_sequence_number"],
+                        lang=sub["lang"],
+                        schedule=sub["schedule"],
+                    )
```

Comment:
Maybe we can rather do a:
```
# There's no service info for SMS
if "service_info" in short_name:
    continue
```
To firstly better explain it, and secondly avoid nesting things too deeply.